### ISSUE TITLE
callsign with prefix

### ISSUE THREAD

**Issue by ec1oud** [kgoba/ft8_lib#44]
I'm American and operating in Norway under the CEPT agreement for now, so I need to use a prefix (my callsign is LA/K7IHZ).  So far, something is wrong with the encoding.  If I add my callsign to the list in test/test.c, the test fails:
```
Encoded [CQ] [LA/K7IHZ] []
Decoded [CQ] [<LA/K7IHZ>] []
FAIL: Condition '0 == strcmp(call_de, call_de_tx)' failed!
...
Encoded [QRZ] [LA/K7IHZ] []
Decoded [QRZ] [<LA/K7IHZ>] []
FAIL: Condition '0 == strcmp(call_de, call_de_tx)' failed!
```
Do the `<>` brackets mean it's always hashed?  

Also, if I use gen_ft8 to encode a wav file with this callsign, `jt9 -8 file.wav` (the tool that comes with wsjt-x) does not successfully decode it, whereas it can decode if the callsign has no prefix.

When testing with ft8code (comes with wsjt-x), it seems like a plain CQ with my callsign but no grid will use the type 4 nonstandard call format, but in many other cases probably it's getting hashed and put into a type 1 standard message.

**Comment by wzab** [comment]
I have added my prefixed and suffixed callsigns to the test: OE/SP5DAA, DE/SP5DAA, SP5DAA/2, SP5DAA/P The lines 166-168 in test.c are now:
```
    const char* callsigns[] = { "YL3JG", "W1A", "W1A/R", "W5AB", "W8ABC", "DE/SP5DAA", "OE/SP5DAA", "SP5DAA/P", "SP5DAA/2", "DE6ABC", "DE6ABC/R", "DE7AB", "DE9A", "3DA0X", "3DA0XYZ", "3DA0XYZ/R", "3XZ0AB", "3XZ0A"};
    const char* tokens[] = { "CQ", "QRZ", "CQ_123", "CQ_000", "CQ_POTA", "CQ_SA", "CQ_O", "CQ_ASD" };
    const char* grids[] = { "KO26", "RR99", "AA00", "RR09", "AA01", "RRR", "RR73", "73", "R+10", "R+05", "R-12", "R-02", "+10", "+05", "-02", "-02", "" };
```
Only SP5DAA/P is encoded and decoded correctly. Three others fail.
 
Below is one of the test cases:

    Encoded [YL3JG] [DE/SP5DAA] [KO26]
    Decoded [YL3JG] [<DE/SP5DAA>] [KO26]
    FAIL: Condition '0 == strcmp(call_de, call_de_tx)' failed!
    Encoded [YL3JG] [OE/SP5DAA] [KO26]
    Decoded [YL3JG] [<OE/SP5DAA>] [KO26]
    FAIL: Condition '0 == strcmp(call_de, call_de_tx)' failed!
    Encoded [YL3JG] [SP5DAA/P] [KO26]
    Decoded [YL3JG] [SP5DAA/P] [KO26]
    Test OK
    Encoded [YL3JG] [SP5DAA/2] [KO26]
    Decoded [YL3JG] [<SP5DAA/2>] [KO26]
    FAIL: Condition '0 == strcmp(call_de, call_de_tx)' failed!
     

 
Another case:

    Encoded [DE/SP5DAA] [YL3JG] [KO26]
    Decoded [<DE/SP5DAA>] [YL3JG] [KO26]
    FAIL: Condition '0 == strcmp(call_to, call_to_tx)' failed!
    Encoded [DE/SP5DAA] [W1A] [KO26]
    Decoded [<DE/SP5DAA>] [W1A] [KO26]
    FAIL: Condition '0 == strcmp(call_to, call_to_tx)' failed!
    Encoded [DE/SP5DAA] [W1A/R] [KO26]
    Decoded [<DE/SP5DAA>] [W1A/R] [KO26]
    FAIL: Condition '0 == strcmp(call_to, call_to_tx)' failed!
    Encoded [DE/SP5DAA] [W5AB] [KO26]

So it seems that the problem still exists, especially for those who are traveling abroad and add DX prefix to their callsign.

**Comment by ec1oud** [comment]
You didn't clarify here that you are using gdyldin's fork though, to get these results, right?

It looks like they all match except for being enclosed in <> brackets?  And AFAIK the brackets just mean that the callsign needed to be hashed.  It would probably be shown that way in the UI of an application that uses ft8_lib, unless the brackets are removed before showing it.

It should be tested against some known-good implementation such as WSJT-X, too.  Encode a wav file with ft8_lib and decode it with WSJT-X, and vice-versa.

I also have doubts, because I was having trouble doing actual QSOs on the air.  And I got feedback from another ham that when I called with "LB2JK/QRP" it got decoded as "LB2JK/" on his end.  So it looks like some cases might be working with that fork, some not.

**Comment by wzab** [comment]
Yes, the results were obtained with the version cloned from https://github.com/gdyuldin/ft8_lib .
